jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3414,18 +3414,30 @@ public String getSimpleName() {
 		}
 	}
 	String simpleName = baseType.getSimpleNameImpl();
+	String fullName = baseType.getName();
+	Class parent = baseType.getEnclosingClass();
```

Comment:
Why the change from `baseType.getEnclosingObjectClass()` (which is a native and doesn't have securitymanager checks) to `baseType.getEnclosingClass`?

Reply:
It looks like `getEnclosingObjectClass()` relies on the `EnclosingMethod` attribute, which means the method returns a non-null value only when the caller is a local or anonymous class. From the JVM spec:

>The EnclosingMethod attribute is a fixed-length attribute in the attributes table of a ClassFile structure (§4.1). A class must have an EnclosingMethod attribute if and only if it represents a local class or an anonymous class (JLS §14.3, JLS §15.9.5).

Instead of calling `getEnclosingClass()`, I changed it to call `getDeclaringClassImpl()` if `parent` is still null when we need it.